envelope.go

```diff
@@ -305,9 +305,6 @@ func (r *envelopeReader) Read(env *envelope) *Error {
 			return NewError(CodeUnknown, err)
 		}
 		err = wrapIfContextError(err)
-		if connectErr, ok := asError(err); ok {
-			return connectErr
-		}
```

Comment:
Not related, but I happened to notice that this was being twice. (On line 312, these exact three lines are repeated. Oops, I think this is from a change of mine.)